src/Raven.Server/Routing/RequestRouter.cs

```diff
@@ -129,6 +135,22 @@ public RouteInformation GetRoute(string method, string path, out RouteMatch matc
             return reqCtx.Database?.Name;
         }
 
+        public static void AssertClientVersion(HttpContext context, Exception innerException)
+        {
+            // client in this context could be also a follower sending a command to his leader.
+            if (context.Request.Headers.TryGetValue(Constants.Headers.ClientVersion, out var versionHeader) && 
+                Version.TryParse(versionHeader, out var clientVersion))
+            {
+                if (clientVersion.Build > ServerVersion.Build)
```

Comment:
As a continuation of:
https://github.com/ravendb/ravendb/pull/6478#discussion_r193503446

Is this correct now? We are sending here strings like `4.1.5.0`. Which means that we are comparing only number `5` here? Which is a semver 'Patch' with `ServerVersion.Build` which is a build number e.g. `40038`.

Reply:
That should be fine here, we explicitly state that we want to allow servers from the same minor version to cooperate together

Reply:
Shouldn't this be changed to:
```csharp
if (clientVersion.Major != RavenVersionAttribute.Instance.AssemblyVersion.Major || clientVersion.Minor != RavenVersionAttribute.Instance.AssemblyVersion.Minor)
```

where `AssemblyVersion` is:

```csharp
        public Version AssemblyVersion
        {
            get
            {
                if (_assemblyVersion == null)
                {
                    var version = typeof(RavenVersionAttribute).GetTypeInfo().Assembly.GetName().Version;
                    _assemblyVersion = new Version(version.Major, version.Minor, version.Build, BuildVersion);
                }

                return _assemblyVersion;
            }
        }
```

?

Reply:
That works